File: leetcode/1.TwoSum.ts
function twoSum(nums: number[], target: number): number[] {
    const hash = new Map<number,number>();
    nums.forEach((x,i) => hash.set(x , i));

    let result: number[] =[0,0];
    for (let i = 0; i < nums.length; i+=1){
        const findValue = target - nums[i];
        const hashValue = hash.get(findValue);
        if (hashValue && i !== hashValue){
            result = [i, hash.get(findValue) as number]
            break;
        }else continue;
    }

    return result;
};

console.log(twoSum([2,7,11,15],9))
console.log(twoSum([3,2,4],6))
console.log(twoSum([3,3],6))

File: leetcode/1.TwoSum.py
def twoSum(nums, target):
    dics = {val: i for i, val in enumerate(nums)}
    for i, val in enumerate(nums):
        find_value = target - val
        if find_value in dics.keys() and i != dics[find_value]:
            return [i, dics[find_value]]

print(twoSum([2,7,11,15],9))
print(twoSum([3,2,4],6))